docs/relax-recursive-type-restriction.md

```diff
@@ -0,0 +1,80 @@
+# Relax the recursive type restriction
+
+## Summary
+
+Relax the [recursive type restriction](recursive-type-restriction.md) by implementing isorecursive
+types properly for parameterized type aliases.
+
+## Motivation
+
+Luau supports parameterized type aliases as follows:
+```luau
+type Box<T> = { data: T }
+```
+
+These type aliases can also be recursive:
+```luau
+type Tree<T> = { data: T, children: {Tree<T>} }
+```
+but the [recursive type restriction](recursive-type-restriction.md) limits us to supporting only
+direct recursion as above, where the type aliases' parameters remain unchanged.
+
+At the time the restriction was implemented, Luau featured a greedy type inference engine that would
+eagerly expand every occurrence of a type alias, and, as such, would infinitely expand until
+ultimately crashing from overflowing the stack, or would hit a recursion limiter and produce a
+strange, very deep type.
```

Comment:
This is not correct, old solver did not behave this way.

Reply:
What did it do then? As far as I could tell from reading old discussions from @asajeffrey, this was the case, but if you know something that I don't, happy to change it.

Reply:
Thanks, I'll revise the text then.